internal/download/artifact.go

```diff
@@ -32,3 +36,43 @@ func Cleanup(directory string) {
 		log.Err(err).Msg("Unable to cleanup previous artifacts")
 	}
 }
+
+// GetDirName returns a target folder name
+func GetDirName(suiteName string, cfg config.ArtifactDownload) (string, error) {
+	suiteName = strings.ReplaceAll(suiteName, "/", "-")
+	suiteName = strings.ReplaceAll(suiteName, "\\", "-")
+	suiteName = strings.ReplaceAll(suiteName, ".", "-")
+	suiteName = strings.ReplaceAll(suiteName, " ", "_")
+	// If targetDir doesn't exist, no need to find maxVersion and return
+	targetDir := filepath.Join(cfg.Directory, suiteName)
+	if _, err := os.Open(targetDir); os.IsNotExist(err) {
+		return targetDir, nil
+	}
+	// Find the maxVersion of downloaded artifacts in artifacts dir
+	f, err := os.Open(cfg.Directory)
+	if err != nil {
+		return "", nil
+	}
+	files, err := f.Readdir(0)
+	if err != nil {
+		return "", err
+	}
+	maxVersion := 0
+	for _, file := range files {
+		if file.IsDir() {
+			fileName := strings.Split(file.Name(), ".")
+			if len(fileName) == 2 && fileName[0] == suiteName {
+				version, err := strconv.Atoi(fileName[1])
+				if err != nil {
+					return "", err
+				}
+				if version > maxVersion {
+					maxVersion = version
+				}
+			}
+		}
+	}
```

Comment:
That's quite the nesting there 😅 

How about:
```suggestion
	for _, file := range files {
		if !file.IsDir() {
			continue
		}

		fileName := strings.Split(file.Name(), ".")
		if len(fileName) != 2 || fileName[0] != suiteName {
			continue
		}

		version, err := strconv.Atoi(fileName[1])
		if err != nil {
			return "", err
		}
		if version > maxVersion {
			maxVersion = version
		}
	}
```